app/full_text_search.py

```diff
@@ -160,17 +160,24 @@ def create_fields_for_rank(field_name, value):
     return fields
 
 
-def create_full_name_without_space(given_name, family_name):
+def create_full_name_without_space(given_names, family_names):
     """
     Creates full name without white space.
     Returns:
-        if given_name and family_name: ('given_name + family_name',
-                                        'family_name + given_name')
+        if given_name and family_name: [('given_name + family_name',
+                                        'family_name + given_name')]
         else: None
     """
-    if given_name and family_name:
-        return (given_name + family_name,
-                family_name + given_name)
+    if given_names and family_names:
+        full_names = []
+        for index1 in xrange(len(given_names)):
+            given_name = given_names[index1]
+            for index2 in xrange(len(family_names)):
+                family_name = family_names[index2]
+                full_names.append(
```

Comment:
How about returning ['given1family1', 'family1given1', 'given1family2', 'family2given1'] instead of [('given1family1', 'family1given1'), ('given1family2', 'family2given1')]? It looks simpler.